web/beacon-app/src/components/layout/MainLayout.tsx

```diff
@@ -3,12 +3,14 @@ import { Outlet } from 'react-router-dom';
 
 import { LandingFooter } from '@/components/auth/LandingFooter';
 import { LandingHeader } from '@/components/auth/LandingHeader';
+import { WelcomePage } from '../ui/WelcomePage';
 
 const MainLayout = () => {
   return (
     <Container>
       <LandingHeader />
       <Outlet />
+      <WelcomePage />
```

Comment:
maybe you could create a router where you can add the welcomePage and remove it from here
``` 
<Route path="/welcome" element={<MainLayout />}>
        <Route
          path="welcome"
          element={<WlecomePage />}
        />
 </Route>
 ```